include/mrdox/Metadata/TemplateParam.hpp

```diff
@@ -13,45 +14,165 @@
 #define MRDOX_METADATA_TEMPLATEPARAM_HPP
 
 #include <mrdox/Platform.hpp>
-#include <clang/AST/Decl.h>
-#include <clang/AST/Type.h>
-#include <llvm/ADT/SmallString.h>
-#include <llvm/ADT/StringRef.h>
+#include <mrdox/Metadata/Type.hpp>
 
 namespace clang {
 namespace mrdox {
 
-// Represents one template parameter.
-//
-// This is a very simple serialization of the text of the source code of the
-// template parameter. It is saved in a struct so there is a place to add the
-// name and default values in the future if needed.
-//
-/** A template parameter.
-*/
+enum class TemplateParamKind : int
+{
+    // KRYSTIAN NOTE: temporary
+    None,
```

Comment:
wtf is "None?" if this is a real thing you need to write
```
None = 0, // for bitstream
```

but if None is going away you need to sort it, I dont like having enums that arent use. For example my `Access` doesn't have `None` (but `clang::AccessSpecifier` does, which is irritating).

Reply:
`None` indicates a `TParam` without any extended info (e.g. `TypeTParam`).